packages/infrastructure-lib/src/mongo_settlement_matrix_repo.ts

```diff
@@ -101,15 +104,53 @@ export class MongoSettlementMatrixRepo implements ISettlementMatrixRequestRepo {
 		}
 	}
 
-	async getMatrices(state?:string): Promise<ISettlementMatrix[]>{
+	async getMatrices(
+		matrixId?: string,
+		type?: string,
+		state?: string,
+		model?: string,
+		currencyCodes?: string[],
+		createdAt?: string,
+		pageIndex: number = 0,
+        pageSize: number = MAX_ENTRIES_PER_PAGE,
+	): Promise<MatrixSearchResults> {
 		try {
-			let resp;
-			if(state){
-				resp = await this._collection.find({state: state}).project({_id: 0}).toArray();
-			}else{
-				resp = await this._collection.find().project({_id: 0}).toArray();
+			// Pagination settings
```

Comment:
In these methods, because we call them in many places in the aggragte, we need the default to be return all (without pagination)